apluslms_roman/configuration.py

```diff
@@ -56,16 +65,61 @@ def find_from(cls, path, name=None, prefixes=None):
                     break
         else:
             if is_legacy_configuration(path):
-                container = cls.Container(config, allow_missing=True)
+                container = cls.Container(join(path, cls.DEFAULT_FILENAME), allow_missing=True)
                 return cls(container, None, LEGACY_CONFIG, cls.version)
-            raise CourseConfigError("Couldn't find course configuration from {}. Expected to find one of these: {}".format(path, ', '.join(files)))
+            raise ProjectConfigError((
+                "Couldn't find course configuration from {}. Expected to find one of these: {}"
+            ).format(path, ', '.join(files)))
 
         return cls.load(config)
 
+    @classmethod
+    def load_from(cls, config):
+        if isfile(config):
+            return cls.load(config)
+        if '.' not in basename(config):
+            for prefix in cls.DEFAULT_PREFIXES:
+                filename = '%s.%s' % (config, prefix)
+                if isfile(filename):
+                    return cls.load(filename)
+        raise ProjectConfigError("Given file '{}' doesn't exist.".format(config))
+
+    def initialize_data(self):
+        super().initialize_data()
+        self.recursive_update(
+            LEGACY_CONFIG if is_legacy_configuration(dirname(self.path))
+            else DEFAULT_CONFIG
+        )
+        self._data[self._version_key] = str(self.version)
+
+    def add_step(self, step):
+        if "name" in step:
+            other_names = [s["name"] for s in [s.get_data() for s in self.steps] if "name" in s]
+            if step["name"] in other_names:
+                raise ValueError("A step with the name '" + step["name"] + "' already exists.")
+        self.steps.append(step)
+
+    def del_step(self, step_ref):
+        step = self.get_step(step_ref)
+        self.steps.remove(step)
+
+    def get_step(self, step_ref):
+        if step_ref.isdigit():
+            return self.steps[int(step_ref)]
+        return self.steps_by_name[step_ref]
+
     @property
     def dir(self):
         return self._container._dir
 
+    @property
+    def path(self):
```

Comment:
Hmm... `Document` has same `.path`, so this should not be needed as it's duplicate, or am I missing someting?

Reply:
...yet another in the series of "stuff I did for some reason but who knows what that reason was"